src/com/amazon/corretto/crypto/provider/keygeneratorspi/SecretKeyGenerator.java

```diff
@@ -0,0 +1,55 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider.keygeneratorspi;
+
+import javax.crypto.KeyGeneratorSpi;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.SecureRandom;
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class SecretKeyGenerator extends KeyGeneratorSpi {
+    private final Supplier<SecureRandom> defaultSecureRandom;
+    private final SecretKeyProperties secretKeyProperties;
+    private Optional<SecureRandom> secureRandom;
+    private int keySize;
+
+    public SecretKeyGenerator(final Supplier<SecureRandom> defaultSecureRandom, final SecretKeyProperties secretKeyProperties) {
+        this.defaultSecureRandom = Objects.requireNonNull(defaultSecureRandom, "defaultSecureRandom cannot be null.");
```

Comment:
There's a degenerate case here where the defaultSecureRandom supplier supplies null objects. Suggest adding a safety check either here, or in `engineGenerateKey` (probably the better place) that asserts that the defaultSecureRandom we get is non-null.

Reply:
I think the `Objects.requireNonNull`s are redundant and I should remove them. This class is instantiated  by ACCP and client code has no control over it. We should statically ensure that the arguments we provide to this constructor are not null rather than adding run-time checks. The same argument goes for the return value of `defaultSecureRandom`, since we provide its implementation.

Reply:
We should start using something like [CheckerFramework](https://checkerframework.org/) to assert that values are not null and there would be no need to add nullness check.